Galley TestAddUpdateDelete Unit Test Failure

Failure: https://prow.k8s.io/view/gcs/istio-prow/logs/istio-unit-tests-master/6413

Also locally found some races

```
==================
WARNING: DATA RACE
Write at 0x00c00035bef0 by goroutine 9:
  istio.io/istio/galley/pkg/config/source/kube/fs.(*source).Stop()
      /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/source/kube/fs/source.go:108 +0x2a9
  istio.io/istio/galley/pkg/config/source/kube/fs_test.TestAddUpdateDelete()
      /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/source/kube/fs/source_test.go:170 +0xb05
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:865 +0x163

Previous read at 0x00c00035bef0 by goroutine 12:
  istio.io/istio/galley/pkg/config/source/kube/fs.(*source).Start.func1()
      /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/source/kube/fs/source.go:89 +0xd2

Goroutine 9 (running) created at:
  testing.(*T).Run()
      /usr/lib/google-golang/src/testing/testing.go:916 +0x65a
  testing.runTests.func1()
      /usr/lib/google-golang/src/testing/testing.go:1159 +0xa8
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:865 +0x163
  testing.runTests()
      /usr/lib/google-golang/src/testing/testing.go:1157 +0x523
  testing.(*M).Run()
      /usr/lib/google-golang/src/testing/testing.go:1074 +0x2f5
  main.main()
      _testmain.go:62 +0x222

Goroutine 12 (finished) created at:
  istio.io/istio/galley/pkg/config/source/kube/fs.(*source).Start()
      /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/source/kube/fs/source.go:79 +0x12f
  istio.io/istio/galley/pkg/config/source/kube/fs_test.startOrFail()
      /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/source/kube/fs/source_test.go:312 +0xc7
  istio.io/istio/galley/pkg/config/source/kube/fs_test.TestAddUpdateDelete()
      /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/source/kube/fs/source_test.go:147 +0x11f
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:865 +0x163
==================
--- FAIL: TestAddUpdateDelete (0.04s)
    source_test.go:143: Root dir: /tmp/configPath346408107
    testing.go:809: race detected during execution of test
FAIL galley/pkg/config/source/kube/fs.TestAddUpdateDelete (0.04s)
=== RUN   TestAddUpdateDelete_K8sResources
2019-08-30T14:55:05.417562Z     info    source  [fs-1] Triggering reload in response to: test
2019-08-30T14:55:05.417886Z     info    source  [fs-1] Discovered file: "/tmp/configPath148253710/bar.yaml"
2019-08-30T14:55:05.427956Z     warn    watcher.Notify: Signal channel is full (trigger: "test", signal: user defined signal 1)
2019-08-30T14:55:05.428077Z     info    source  [fs-1] Triggering reload in response to: test
2019-08-30T14:55:05.428325Z     info    source  Removing the contents of the file "/tmp/configPath148253710/bar.yaml"
==================
WARNING: DATA RACE
Write at 0x00c00047a7b0 by goroutine 16:
  istio.io/istio/galley/pkg/config/source/kube/fs.(*source).Stop()
      /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/source/kube/fs/source.go:108 +0x2a9
  istio.io/istio/galley/pkg/config/source/kube/fs_test.TestAddUpdateDelete_K8sResources()
      /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/source/kube/fs/source_test.go:207 +0x1999
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:865 +0x163

Previous read at 0x00c00047a7b0 by goroutine 17:
  istio.io/istio/galley/pkg/config/source/kube/fs.(*source).Start.func1()
      /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/source/kube/fs/source.go:89 +0xd2

Goroutine 16 (running) created at:
  testing.(*T).Run()
      /usr/lib/google-golang/src/testing/testing.go:916 +0x65a
  testing.runTests.func1()
      /usr/lib/google-golang/src/testing/testing.go:1159 +0xa8
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:865 +0x163
  testing.runTests()
      /usr/lib/google-golang/src/testing/testing.go:1157 +0x523
  testing.(*M).Run()
      /usr/lib/google-golang/src/testing/testing.go:1074 +0x2f5
  main.main()
      _testmain.go:62 +0x222

Goroutine 17 (finished) created at:
  istio.io/istio/galley/pkg/config/source/kube/fs.(*source).Start()
      /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/source/kube/fs/source.go:79 +0x12f
  istio.io/istio/galley/pkg/config/source/kube/fs_test.startOrFail()
      /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/source/kube/fs/source_test.go:312 +0xc7
  istio.io/istio/galley/pkg/config/source/kube/fs_test.TestAddUpdateDelete_K8sResources()
      /usr/local/google/home/howardjohn/go/src/istio.io/istio/galley/pkg/config/source/kube/fs/source_test.go:179 +0x133
  testing.tRunner()
      /usr/lib/google-golang/src/testing/testing.go:865 +0x163
==================

```

--- a/galley/pkg/config/source/kube/fs/source.go
+++ b/galley/pkg/config/source/kube/fs/source.go
@@ -87,7 +87,7 @@ func (s *source) Start() {
 					scope.Source.Infof("[%s] Triggering reload in response to: %v", s.name, trigger.Source)
 					s.reload()
 				}
-			case <- done:
+			case <-done:
 				return
 			}
 		}

--- a/galley/pkg/config/source/kube/fs/source.go
+++ b/galley/pkg/config/source/kube/fs/source.go
@@ -72,7 +72,8 @@ func (s *source) Start() {
 	if s.done != nil {
 		return
 	}
-	s.done = make(chan struct{})
+	done := make(chan struct{})
+	s.done = done
 
 	c := make(chan appsignals.Signal, 1)
 	appsignals.Watch(c)
@@ -86,7 +87,7 @@ func (s *source) Start() {
 					scope.Source.Infof("[%s] Triggering reload in response to: %v", s.name, trigger.Source)
 					s.reload()
 				}
-			case <-s.done:
+			case <- done:
 				return
 			}
 		}

--- a/galley/pkg/config/testing/fixtures/expect.go
+++ b/galley/pkg/config/testing/fixtures/expect.go
@@ -47,7 +47,7 @@ func ExpectFilter(t *testing.T, acc *Accumulator, fn FilterFn, expected ...event
 	g := gomega.NewGomegaWithT(t)
 
 	wrapFn := func() []event.Event {
-		e := acc.events
+		e := acc.Events()
 		if fn != nil {
 			e = fn(e)
 		}

